calico_kubernetes/calico_kubernetes.py

```diff
@@ -118,12 +118,24 @@ def create(self, namespace, pod_name, docker_id):
             endpoint = self._configure_interface()
             logger.info("Created Calico endpoint: %s", endpoint.endpoint_id)
             self._configure_profile(endpoint)
-        except CalledProcessError as e:
-            logger.error('Error code %d creating pod networking: %s\n%s',
-                         e.returncode, e.output, e)
+        except BaseException:
+            # Check to see if an endpoint has been created.  If so,
+            # we need to tear down any state we may have created.
+            logger.exception("Error networking pod - cleaning up")
+
+            try:
+                self.delete(namespace, pod_name, docker_id)
```

Comment:
Does .delete() make a best effort to reverse all of the side effects of everything in the try: block? From a quick code read it looks like it doesn't. If that's OK, I think you need to add a comment to justify why.